openedx/core/djangoapps/user_api/views.py

```diff
@@ -837,6 +837,30 @@ def _add_terms_of_service_field(self, form_desc, required=True):
             supplementalText=terms_text
         )
 
+    def _add_request_data_sharing_consent_field(self, form_desc, provider):
+        default_label = _(u"I consent to share coursework data with {provider_name}").format(
+            provider_name=provider.name
+        )
+        label = provider.data_sharing_consent_prompt or default_label
+
+        default_error_msg = _(u"{provider_name} requires that you consent to sharing coursework data.").format(
+            provider_name=provider.name
```

Comment:
This one is a tough call - while English grammar does require a capitalized word at the beginning of a sentence, a name might be stylized as lowercase, which would override that - for example, "iPhone requires you to enter your password" or something like that.